implementation/demos/compare_cuas_snes.py

```diff
@@ -0,0 +1,164 @@
+                                help="Displacement BC in negative y direction")
+    _ref = parser.add_argument("--refinements", default=1, type=np.int32,
+                               dest="refs", help="Number of mesh refinements")
+    _gap = parser.add_argument(
+        "--gap", default=0.02, type=np.float64, dest="gap", help="Gap between plane and y=0")
+
+    # Parse input arguments or set to defualt values
+    args = parser.parse_args()
+
+    # Current formulation uses unilateral contact
+    nitsche_parameters = {"gamma": args.gamma, "theta": args.theta}
+    nitsche_bc = not args.dirichlet
+    physical_parameters = {"E": args.E, "nu": args.nu, "strain": args.plane_strain}
+    vertical_displacement = -args.disp
+    num_refs = args.refs + 1
+    gap = args.gap
+    top_value = 1
+    threed = args.threed
+    bottom_value = 2
+
+    # Load mesh and create identifier functions for the top (Displacement condition)
+    # and the bottom (contact condition)
+    if threed:
+        fname = "sphere"
+        create_sphere_mesh(filename=f"{fname}.msh")
+        convert_mesh(fname, "tetra")
+        with dolfinx.io.XDMFFile(MPI.COMM_WORLD, f"{fname}.xdmf", "r") as xdmf:
+            mesh = xdmf.read_mesh(name="Grid")
+        # mesh = dolfinx.UnitCubeMesh(MPI.COMM_WORLD, 10, 10, 20)
+
+        # def top(x):
+        #     return x[2] > 0.99
+
+        # def bottom(x):
+        #     return x[2] < 0.5
+        def top(x):
+            return x[2] > 0.9
+
+        def bottom(x):
+            return x[2] < 0.15
+
+    else:
+        fname = "disk"
+        create_disk_mesh(filename=f"{fname}.msh")
+        convert_mesh(fname, "triangle", prune_z=True)
+        with dolfinx.io.XDMFFile(MPI.COMM_WORLD, f"{fname}.xdmf", "r") as xdmf:
+            mesh = xdmf.read_mesh(name="Grid")
+        # mesh = dolfinx.UnitSquareMesh(MPI.COMM_WORLD, 30, 30)
+
+        # def top(x):
+        #     return x[1] > 0.99
+
+        def top(x):
+            return x[1] > 0.5
+
+        def bottom(x):
+            return x[1] < 0.2
+
+    e_abs = []
+    e_rel = []
+    dofs_global = []
+    rank = MPI.COMM_WORLD.rank
+    refs = np.arange(0, num_refs)
+    for i in refs:
+        if i > 0:
+            # Refine mesh
+            mesh.topology.create_entities(mesh.topology.dim - 2)
+            mesh = dolfinx.mesh.refine(mesh)
+
+        # Create meshtag for top and bottom markers
+        tdim = mesh.topology.dim
+        top_facets = dolfinx.mesh.locate_entities_boundary(mesh, tdim - 1, top)
+        bottom_facets = dolfinx.mesh.locate_entities_boundary(
+            mesh, tdim - 1, bottom)
+        top_values = np.full(len(top_facets), top_value, dtype=np.int32)
+        bottom_values = np.full(
+            len(bottom_facets), bottom_value, dtype=np.int32)
+        indices = np.concatenate([top_facets, bottom_facets])
+        values = np.hstack([top_values, bottom_values])
+        facet_marker = dolfinx.MeshTags(mesh, tdim - 1, indices, values)
```

Comment:
There should be a sorting command here, otherwise this will break in parallel, see: https://jorgensd.github.io/dolfinx-tutorial/chapter2/hyperelasticity.html?highlight=argsort